wgsl/index.bs

```diff
@@ -422,6 +422,147 @@ The result type depends on the number of letters provided. Assuming a `vec4<f32>
+
+`texture_wo_1d_array<image_storage_type>`
+  %1 = OpTypeImage %void 1D 0 1 0 2 image_storage_type
+
+`texture_wo_2d<image_storage_type>`
+  %1 = OpTypeImage %void 2D 0 0 0 2 image_storage_type
+
+`texture_wo_2d_array<image_storage_type>`
+  %1 = OpTypeImage %void 2D 0 1 0 2 image_storage_type
+
+`texture_wo_3d<image_storage_type>`
+  %1 = OpTypeImage %void 3D 0 0 0 2 image_storage_type
+</pre>
+
+### Depth Texture ### {#texture-depth}
+<pre class='def'>
+`texture_depth_2d`
+  %1 = OpTypeImage %f32 2D 1 0 0 1 Unknown
+
+`texture_depth_2d_array`
+  %1 = OpTypeImage %f32 2D 1 1 0 1 Unknown
+</pre>
+
+### Samplers ### {#texture-sampler}
+<pre class='def'>
+sampler
+  OpTypeSampler
+
+sampler_comparison
+  OpTypeSampler
+</pre>
+
+## Texture Functions ## {#texture-functions}
+
+<pre class='def'>
+`vec4<type> texture_load(texture_ro, i32 coords, i32 level_of_detail)`
+`vec4<type> texture_load(texture_ro, vec2<i32> coords, i32 level_of_detail)`
+`vec4<type> texture_load(texture_ro, vec3<i32> coords, i32 level_of_detail)`
+  %32 = OpImageFetch %v4float %texture_ro %coords Lod %level_of_detail
+
+TODO(dsinclair): Texture load on a sampled texture
+
+TODO(dsinclair): Texture load on multi-sampled texture
+
+TODO(dsinclair): Add `texture_write` method for texture_wo with integral coords
+
+TODO(dsinclair): Allow a small constant offset on the coordinate? May not be portable.
+
+`vec4<type> texture_sample(texture_sampled, sampler, f32 coords)`
+`vec4<type> texture_sample(texture_sampled, sampler, vec2<f32> coords)`
+`vec4<type> texture_sample(texture_sampled, sampler, vec3<f32> coords)`
+`vec4<type> texture_sample(texture_sampled, sampler, vec4<f32> coords)`
+  %24 = OpImageSampleImplicitLod %v4float %sampled_image %coords
+
+`vec4<type> texture_sample_level(texture_sampled, sampler, f32 coords, f32 lod)`
+`vec4<type> texture_sample_level(texture_sampled, sampler, vec2<f32> coords, f32 lod)`
+`vec4<type> texture_sample_level(texture_sampled, sampler, vec3<f32> coords, f32 lod)`
+`vec4<type> texture_sample_level(texture_sampled, sampler, vec4<f32> coords, f32 lod)`
+  %25 = OpImageSampleExplicitLod %v4float %sampled_image %coords Lod %lod
+
+`vec4<type> texture_sample_bias(texture_sampled, sampler, f32 coords, f32 bias)`
+`vec4<type> texture_sample_bias(texture_sampled, sampler, vec2<f32> coords, f32 bias)`
+`vec4<type> texture_sample_bias(texture_sampled, sampler, vec3<f32> coords, f32 bias)`
+`vec4<type> texture_sample_bias(texture_sampled, sampler, vec4<f32> coords, f32 bias)`
+  %19 = OpImageSampleImplicitLod %v4float %sampled_image %coords Bias %bias
+
+`f32 texture_sample_compare(texture_depth, sampler_comparison, f32 coords, f32 depth_reference)`
+`f32 texture_sample_compare(texture_depth, sampler_comparison, vec2<f32> coords, f32 depth_reference)`
+`f32 texture_sample_compare(texture_depth, sampler_comparison, vec3<f32> coords, f32 depth_reference)`
+  %65 = OpImageSampleDrefImplicitLod %float %sampled_image %coord %depth_reference
```

Comment:
I think this should be `OpImageSampleDrefExplicitLod` with lod = 0.0

Reply:
Done